### PR TITLE
[#522] All courses delete section added (Connects #522)

### PR DESCRIPTION
**Pull request by BasnetSamriddhi**

### CODE REVIEW

File: src/app/courses/courses.component.ts [hunk 1]
```diff
@@ -49,12 +49,10 @@ export class CoursesComponent implements OnInit, AfterViewInit {
   getCourses() {
     this.couchService.get('courses/_all_docs?include_docs=true')
       .subscribe((data) => {
-        this.courses.data = data.rows.map((course: any) => {
-          return course.doc;
-        }).filter((c: any) => {
-          return c._id !== '_design/course-validators';
-        });
-      }, (error) => this.planetMessageService.showAlert('There was a problem getting courses'));
+         // Sort in descending articleDate order, so the new courses can be shown on the top
```
Comment (lmmrssa) [inline src/app/courses/courses.component.ts]: this code does not seem to be related to this issue

File: src/app/resources/resources.component.ts [hunk 2]
```diff
@@ -178,7 +178,7 @@ export class ResourcesComponent implements OnInit, AfterViewInit {
           this.getResources();
           this.selection.clear();
           this.deleteDialog.close();
-          this.planetMessageService.showAlert('You have deleted all resources');
```
Comment (paulbert) [inline src/app/resources/resources.component.ts]: Please remove this change.  It's unrelated to the issue.

File: src/app/courses/courses.component.ts [hunk 3]
```diff
@@ -72,14 +72,36 @@ export class CoursesComponent implements OnInit, AfterViewInit {
...
+      displayName = course.courseTitle;
+    }
+    this.openDeleteDialog(okClick, amount, displayName);
+  }
+
+  openDeleteDialog(okClick, amount, displayName = '') {
     this.deleteDialog = this.dialog.open(DialogsPromptComponent, {
       data: {
-        okClick: this.deleteCourse(course),
+        okClick,
```
Comment (lmmrssa) [inline src/app/courses/courses.component.ts]: this is just the value being set, it should be key value pair
